install.md

```diff
@@ -1,63 +0,0 @@
-# conmon Installation Instructions
```

Comment:
I think we had a reason to put the content into a dedicated `install.md`, right @haircommander @TomSweeneyRedHat ?

Reply:
Oh I don't understand if any latent needs so just simply clean it up together (please kindly correct me) ;-)

Reply:
:shrug: I don't remember exactly why we went for an install.md, TBH. a reference to it in the README would probably be good in general. I'm fine with either approach

Reply:
Seems no specific reason: https://github.com/containers/conmon/pull/61